lib/voucherify/service/validations.rb

```diff
@@ -12,7 +12,16 @@ def initialize(client)
       def validate_voucher(code, context = {})
         @client.post("/vouchers/#{URI.encode(code)}/validate", context.to_json)
       end
-      
+
+      def validate(code, context = {})
+        if code.is_a? String
```

Comment:
same here, first let's check for a hash and then just assume it is a String and execute appropriate calls